tests/literal.json

```diff
@@ -132,6 +132,41 @@
           "comment": "Literal on RHS of subexpr not allowed",
           "expression": "foo.`\"bar\"`",
           "error": "syntax"
+        },
+        {
+          "comment": "Literal with all valid leading and trailing JSON whitespace",
+          "expression": "` \t\n\r{\"foo\": true} \t\n\r`",
+          "result": {"foo": true}
+        },
+        {
+          "comment": "Literal with non-JSON whitespace U+000C FORM FEED",
```

Comment:
Shouldn't the "literal with non-JSON whitespace" test cases be moved into their own file so implementations that support the legacy syntax can filter them out per [JEP-12: Impact](https://github.com/jmespath-community/jmespath.spec/blob/main/jep-012-raw-string-literals.md#impact) (emphasis mine)?
> In order to support this type of variance in JMESPath implementations, all of the JSON literal compliance test cases that involve elided quotes MUST be removed, and **test cases regarding failing on invalid unquoted JSON values MUST not be allowed in the compliance test unless placed in a JEP 12 specific test suite**, allowing implementations that support elided quotes in JSON literals to filter out the JEP 12 specific test cases.

I think we want to end up in a situation where any given test run includes either the legacy test cases or the syntax-error test cases but never both—or alternatively that the test cases are exactly the same but the expected results are configurable, e.g.
```json
        {
          "comment": "Literal with non-JSON whitespace U+000C FORM FEED",
          "expression": "`0\f`",
          "error": "syntax",
          "jep12LegacyResult": "0\f",
        },
```

Reply:
Hum that's a good point. However I did not want to dwell too long on expanding the existing legacy test suite that James already isolated.

The way I intend this folder organization is that common tests should be run with enforced standards compliance.

Tests in the legacy folder are intended for implementations that opt in to the old behaviour.

Reply:
That's how I plan the [upcoming changes to the Python implementation](https://github.com/jmespath-community/jmespath.py/tree/feature/vnext-preview/tests) to work.

See https://github.com/jmespath-community/jmespath.py/blob/69f3f9487e2360727e1252d21b24a367835464f9/tests/test_compliance.py#L74-L76

Reply:
If I understand that code correctly, the Python implementation testing is running all tests, but using legacy behavior for legacy test files and non-legacy behavior for all other test files. That will definitely work, but it means that any implementation which _doesn't_ have (configurable) non-legacy behavior must either skip the file completely (missing out on unrelated coverage) or add a custom approach for detecting failure to raise the expected syntax error on these specific (unmarked) test cases. Wouldn't it be better to instead reduce the burden on those implementations by isolating the test cases into their own file as suggested in JEP-12?

Reply:
Evidently, you were right.